Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -415,10 +417,10 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
             }
         }
     }
-    if (/*self.statusCode == 401 // TODO: check for status code once the bug returning 500 is fixed
-         &&*/ client.accessToken.refreshToken != nil
-        && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+    
+    BOOL shouldRefresh = (self.statusCode == 401) && (client.accessToken.hasExpired) && (client.accessToken.refreshToken != nil);
```

Comment:
I'm for using a BOOL it makes things more clear. As for the commented out code, that was an oversight in another merge and it's removed in https://github.com/nxtbgthng/OAuth2Client/commit/5fe6147b5e1ff81b9d3eb1fad5bb9e085a38a7bb